hal/src/thumbv7em/usb/bus.rs

```diff
@@ -800,16 +789,33 @@ impl Inner {
                     //  instead it is cleared in the read handler.
```

Comment:
Does this comment block contradict the core change of this PR?

Reply:
Interesting question!  I think you're correct that it does in spirit, so I'll reword it.

The setup bit is only relevant for control endpoints, and I believe these are nearly-always endpoint 0 in the real world - usb-device currently only supports control pipes on EP0 anyway.  The [usb-device poll()](https://github.com/mvirkkunen/usb-device/blob/b6037b0f98c933ff0a18a864fd34a42f89a6316a/src/device.rs#L206-L207) (which calls this code) immediately handles control transfers on endpoint 0, calling `read()` in the process, so there's practically no difference from immediately clearing the setup interrupt flag.